### PR TITLE
feat: add script to transfer object

### PR DESCRIPTION
**Pull request by blockchainguyy**
https://axelarnetwork.atlassian.net/browse/AXE-4132

### CODE REVIEW

**Review by re1ro — APPROVED** [review]

**Review by npty — APPROVED** [review]

File: sui/transfer-object.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+if (require.main === module) {
+    const program = new Command();
+
+    program.name('transfer-object').description('Transfer object to recipient address');
+
+    addBaseOptions(program);
+    program.addOption(new Option('--objectId <objectId>', 'object id to be transferred').env('SUI_OBJECT_ID'));
+    program.addOption(new Option('--contractName <contractName>', 'contract name').env('CONTRACT_NAME'));
+    program.addOption(new Option('--objectName <objectName>', 'object name to be transferred').env('SUI_OBJECT_NAME'));
```
Comment (milapsheth) [inline sui/transfer-object.js:67]: let's avoid env vars declarations unless needed to avoid polluting env capture

File: sui/transfer-object.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    program.addOption(
+        new Option('--recipient <recipientAddress>', 'recipient to transfer object to').makeOptionMandatory(true).env('RECIEPENT_ADDRESS'),
```
Comment (milapsheth) [inline sui/transfer-object.js:69]:
```suggestion
        new Option('--recipient <recipient>', 'recipient to transfer object to').makeOptionMandatory(true),
```
match flag, and to keep it shorter

File: sui/transfer-object.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        throw new Error('provide object id or contract name with object name');
+    }
+}
+
+if (require.main === module) {
+    const program = new Command();
+
+    program.name('transfer-object').description('Transfer object to recipient address');
+
+    addBaseOptions(program);
```
Comment (milapsheth) [inline sui/transfer-object.js:64]:
```suggestion
    addBaseOptions(program);

```

File: sui/transfer-object.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    program.addOption(new Option('--objectId <objectId>', 'object id to be transferred').env('SUI_OBJECT_ID'));
+    program.addOption(new Option('--contractName <contractName>', 'contract name').env('CONTRACT_NAME'));
```
Comment (milapsheth) [inline sui/transfer-object.js:66]: create extended options similar to EVM to add contractName, will be useful for other scripts. We should also merge EVM / Sui CLI helpers since there's a lot of reuse in a separate task

File: sui/transfer-object.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    program.addOption(new Option('--objectName <objectName>', 'object name to be transferred').env('SUI_OBJECT_NAME'));
+    program.addOption(
+        new Option('--recipient <recipientAddress>', 'recipient to transfer object to').makeOptionMandatory(true).env('RECIEPENT_ADDRESS'),
+    );
+
+    program.action(async (options) => {
+        await main(options);
```
Comment (milapsheth) [inline sui/transfer-object.js:73]: use the `mainProcessor` pattern as done in other sui scripts (and evm)

File: sui/transfer-object.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            showObjectChanges: true,
+            showBalanceChanges: true,
+            showEvents: true,
+        },
+    });
+    printInfo('Transaction result', JSON.stringify(result));
+}
+
+async function getConfigAndTransferObject(options) {
+    const objectId = loadConfig(options.env).sui.contracts[`${options.contractName}`].objects[`${options.objectName}`];
```
Comment (milapsheth) [inline sui/transfer-object.js:28]: not very readable, split into multiple lines, use `?` operator to handle undefined cases, and throw error. Also, config loading is being duplicated. No reason to have transferCap in a different function, while this just retrieves objectId from config. See the `processCommand` pattern from other scripts.

File: sui/transfer-object.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    await transferCap(options);
+}
+
+async function main(options) {
+    if (!isKeccak256Hash(options.recipient)) {
+        throw new Error(`Invalid recipient [${options.recipient}]`);
+    }
+
+    if (options.objectId) {
+        if (!isKeccak256Hash(options.objectId)) {
```
Comment (milapsheth) [inline sui/transfer-object.js:39]: see validateParameters helper from other scripts for simpler validation

File: sui/utils.js [hunk 8]
```diff
@@ -7,6 +7,7 @@ const {
     utils: { arrayify },
 } = ethers;
 const { CosmWasmClient } = require('@cosmjs/cosmwasm-stargate');
+require('dotenv').config();
```
Comment (milapsheth) [inline sui/utils.js:10]: already done in `cli-utils`, not needed here

**Review by milapsheth — CHANGES_REQUESTED** [review]

**Review by Foivos — APPROVED** [review]

**Review by re1ro — APPROVED** [review]

File: sui/transfer-object.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+        if (!objectsData) {
+            throw new Error(`contract objects not found`);
+        }
+
+        objectId = objectsData[`${objectName}`];
+
+        if (!objectId) {
+            throw new Error(`object name [${objectName}] not found`);
+        }
```
Comment (milapsheth) [inline sui/transfer-object.js:51]: Very verbose, use the optional chaining operator to get the objectId directly from chain. E.g. `chain.contracts?.[contractName]...` and validate the objectId at the end outside the if blocks generically. See our usage of ?. operator in other scripts.

File: sui/transfer-object.js [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    const recipient = options.recipient;
+    validateParameters({
+        isKeccak256Hash: { recipient },
+    });
+
+    let objectId;
+
+    if (options.objectId) {
+        objectId = options.objectId;
+        validateParameters({
```
Comment (milapsheth) [inline sui/transfer-object.js:20]: Both branches need to check the objectId, so do it at the end and avoid duplication

File: sui/transfer-object.js (hunk 9 above)
Comment (milapsheth) [reply sui/transfer-object.js]: no, it was just for readability of the intermediate objects, since the line was quite long and complex. `?` simplifies validation checks

File: sui/transfer-object.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+const { printInfo, validateParameters } = require('../evm/utils');
+const { addExtendedOptions } = require('./cli-utils');
+const { getWallet, printWalletInfo } = require('./sign-utils');
+const { loadSuiConfig } = require('./utils');
+
+async function processCommand(chain, options) {
+    const [keypair, client] = getWallet(chain, options);
+    await printWalletInfo(keypair, client, chain, options);
+    const recipient = options.recipient;
+    validateParameters({
```
Comment (milapsheth) [inline sui/transfer-object.js:12]:
```suggestion

    validateParameters({
```

File: sui/transfer-object.js [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+        const contractsData = chain?.contracts;
+        const contractObject = contractsData?.[contractName];
+        const objectsData = contractObject?.objects;
+        objectId = objectsData?.[objectName];
+    } else {
+        throw new Error('Provide object id or contract name with object name');
+    }
+
+    validateParameters({
+        isNonEmptyString: { objectId },
```
Comment (milapsheth) [inline sui/transfer-object.js:36]: why is this needed if we're checking the below?

**Review by milapsheth — APPROVED** [review]
